job-scheduler/src/main/java/org/hawkular/metrics/scheduler/impl/JobsService.java

```diff
@@ -66,9 +68,19 @@
 
     private PreparedStatement updateJobParameters;
 
+    private PreparedStatement findByIdAndSlice;
+
     private static final Function<Map<String, String>, Completable> SAVE_PARAMS_NO_OP =
             params -> Completable.complete();
 
+    // Fields that shouldn't be null: time_slice, job_id, job_type, job_name, trigger
+    private Func1<Row, Boolean> filterAllScheduledQuery = row -> !(row.isNull(0) || row.isNull(1)
+            || row.isNull(2) || row.isNull(3) || row.isNull(5));
+
```

Comment:
The functions look good. I'll mention one thing to consider. The filters should help avoid the NPEs but will do so silently. You could add some additional logging at the expense of making the filter functions a bit more verbose, e.g., 

```java
private Func1<Row, Boolean> filterAllScheduledQuery = row -> {
    boolean isNull = !(row.isNull(0) || row.isNull(1) || row.isNull(2) || row.isNull(3) || row.isNull(5));
    if (isNull) {
        // log column names/values...
    }
    return isNull;
}
```